standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -132,10 +132,12 @@ class CompactionTxnHandler extends TxnHandler {
               " OR (\"res1\".\"TC_PARTITION\" IS NULL AND \"res3\".\"CQ_PARTITION\" IS NULL))" +
       " WHERE \"res3\".\"RETRY_RECORD_CHECK\" <= 0 OR \"res3\".\"RETRY_RECORD_CHECK\" IS NULL";
 
-  private static final String DELETE_ABORT_RETRY_ENTRIES_FROM_COMPACTION_QUEUE =
-      "DELETE FROM \"COMPACTION_QUEUE\" WHERE \"CQ_DATABASE\" = ? " +
-      " AND \"CQ_TABLE\" = ? AND (\"CQ_PARTITION\" = ? OR \"CQ_PARTITION\" IS NULL) AND \"CQ_TYPE\" = " +
-      quoteChar(TxnStore.ABORT_TXN_CLEANUP_TYPE);
+  private static final String DELETE_RETRY_ENTRIES = "DELETE FROM \"COMPACTION_QUEUE\" WHERE \"CQ_ID\" = ?";
+
+  private static final String DELETE_COMPACTION_AND_RETRY_ENTRIES = DELETE_RETRY_ENTRIES +
+      " OR ( \"CQ_DATABASE\" = ? AND \"CQ_TABLE\" = ? " +
+      " AND \"CQ_PARTITION\" %s AND \"CQ_TYPE\" = " +
+      quoteChar(TxnStore.ABORT_TXN_CLEANUP_TYPE) + " )";
```

Comment:
private static final String DELETE_CQ_AND_ABORT_ENTRIES = DELETE_RETRY_ENTRIES +
      " OR ( \"CQ_DATABASE\" = ? AND \"CQ_TABLE\" = ?  AND \"CQ_PARTITION\" %s"
      " AND \"CQ_TYPE\" = " + quoteChar(TxnStore.ABORT_TXN_CLEANUP_TYPE) + " )";

Reply:
Done.